src/App.jsx

```diff
@@ -2,20 +2,29 @@ import React, { Fragment, useEffect, useState } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState(key, initialState) {
-  const [value, setValue] = useState(
-    JSON.parse(localStorage.getItem(key)) || initialState
-  );
+function App() {
+  const key = 'savedTodoList';
+  const initialState = JSON.parse(localStorage.getItem(key)) || [];
+  const [todoList, setTodoList] = useState(initialState);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    localStorage.setItem(key, JSON.stringify(value));
-  }, [key, value]);
-
-  return [value, setValue];
-}
-
-function App() {
-  const [todoList, setTodoList] = useSemiPersistentState('savedTodoList', []);
+    const fetchData = async () => {
+      try {
+        // Simular una llamada a una API o un proceso de carga de datos
+        await new Promise((resolve) => setTimeout(resolve, 2000));
+        
+        // Una vez que los datos se cargan con éxito, actualizamos el estado
+        setTodoList(initialState);
+        setIsLoading(false);
+      } catch (error) {
+        console.error('Error al cargar los datos:', error);
+        setIsLoading(false); // En caso de error, también detenemos el indicador de carga
+      }
+    };
+
+    fetchData();
+  }, [initialState]);
```

Comment:
There's a bug in your code, if you add a new todo, the fetchData function is gonna rerun each time the savedTodoList is updated in local storage, because you have it as an array dependency for the useEffect, but the best approach is remove that dependency and have an empty array (so it'll run once in mounting) in this case it's an expected behaviour of your app: When the app starts you want to fetch the data from an API (in this case a simulation with the promise and setTimeout)